source/fundamentals/poco.txt

```diff
@@ -0,0 +1,273 @@
+   :copyable: true
+
+   public class House
+   {
+       public ObjectId Id { get; set; }
+
+       [BsonRepresentation(BsonType.Int32)]
+       public char YearBuilt { get; set; }
+   }
+
+Set Field Order
+~~~~~~~~~~~~~~~
+
+The driver serializes properties to BSON fields in the order they
+are specified in the POCO. To store properties in a custom order to
+match an existing schema, you can specify the ``Order`` named
+parameter in the ``[BsonElement()]`` attribute. In the following code
+sample, the driver stores the ``YearBuilt`` property after the
+``Style`` property:
+
+.. code-block:: csharp
+   :copyable: true
+
+   public class House
+   {
+       public ObjectId Id { get; set; }
+
+       [BsonElement(Order = 2)]
+       public int YearBuilt { get; set; }
+
+       [BsonElement(Order = 1)]
+       public string Style { get; set; }
+   }
+
+If any properties don't have an explicit ``Order``, the driver will
+serialize them in the default order after those that do.
+
+Identify ``Id`` Property
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, the driver maps any public property named ``Id``, ``id``, or
+``_id`` to the BSON ``_id`` field. To explicitly select the
+property to map to the ``_id`` field, use the ``[BsonId()]`` attribute.
+The following code sample maps the ``Identifier`` property to the
+``_id`` field:
+
+.. code-block:: csharp
+   :copyable: true
+
+   public class House
+   {
+       [BsonId]
+       public string Identifier { get; set; }
+   }
+
+.. warning:: Multiple Id Fields
+
+   If you identify more than one property as the ``_id`` field, the
+   driver throws a ``BsonSerializationException``.
+
+Omit Empty Fields
+~~~~~~~~~~~~~~~~~
+
+By default, the driver serializes undefined properties to fields with ``null``
+values. To ignore undefined properties during serialization, use the ``[BsonIgnore()]``
+attribute. The following code shows how you can prevent the driver from
+serializing the ``YearBuilt`` property if it is undefined:
+
+.. code-block:: csharp
+   :copyable: true
+
+   public class House
+   {
+       public ObjectId Id { get; set; }
+
+       [BsonIgnore]
+       public int YearBuilt { get; set; }
+       public string Style { get; set; }
+   }
+
+Example
+-------
+
+The following example shows how to define a {+language+} class with custom field
+mapping specifications, create a POCO to represent a document, then
+insert the document into MongoDB. The ``Clothing`` class uses
+serialization attributes to perform the following:
+
+- Sets a custom field name ``StyleName`` for the ``Name`` property
+- Serializes the ``Price`` field as a BSON ``Double`` type
+
+Define the ``Clothing`` class with the relevant serialization-related
+attributes:
+
+.. literalinclude:: ../includes/fundamentals/code-examples/poco.cs
+   :start-after: start-model
+   :end-before: end-model
+   :language:  csharp
+   :copyable:
+   :dedent:
+
+Instantiate a ``Clothing`` object and insert the document into a collection:
```

Comment:
```suggestion
The `Program` class instantiates a ``Clothing`` object and inserts the document into a collection:
```

Reply:
reworded this and the preceding code intro to say "The following code..."